onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -59,7 +59,8 @@ Status AttentionBase::CheckInputs(const TensorShape& input_shape,
   //   input       : (batch_size, sequence_length, hidden_size)
   //   weights     : (hidden_size, 3 * hidden_size)
   //   bias        : (3 * hidden_size)
-  //   mask_index  : nullptr, (batch_size), (2 * batch_size), (batch_size, 1), (1, 1) or (batch_size, past_sequence_length + sequence_length)
+  //   mask_index  : nullptr, (batch_size), (2 * batch_size), (batch_size, 1), (1, 1), (batch_size, past_sequence_length + sequence_length)
+  //                 or (batch_size, 1, past_sequence_length + sequence_length)
```

Comment:
Since the final mask data is (B, N, S, S*), so 3D input mask is better to be (B, S, S*) or (batch_size, sequence_length, past_sequence_length+sequence_length).

For (B, 1, S*), we can add a squeeze node to convert it to (B, S*) in fusion, and no need to change the operator.

Reply:
added